### PR TITLE
Don't lint `single_component_path_imports` in macros

### LINKED ISSUE #5154: Redundant Import Warning on Diesel Items
I'm currently running Clippy against a project I've made using Rocket and Diesel, and I'm getting redundant import warnings, specifically against items from Diesel. 

```
warning: this import is redundant
  --> src/db/auth_repository.rs:10:10
   |
10 | #[derive(Insertable)]
   |          ^^^^^^^^^^
   |
   = note: `#[warn(clippy::single_component_path_imports)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports

warning: this import is redundant
 --> src/models/user.rs:7:10
  |
7 | #[derive(Queryable, Serialize)]
  |          ^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports

warning: this import is redundant
 --> src/schema.rs:1:1
  |
1 | / table! {
2 | |     users (id) {
3 | |         id -> Int4,
4 | |         first_name -> Text,
... |
8 | |     }
9 | | }
  | |_^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_component_path_imports
  = note: this warning originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

I could somehow be misunderstanding, but I'm not sure how a derived attribute could be a redundant import, nor how the schema could be. I stepped away from working on this project for a few months, so I'm not sure when this started, as I only updated all of my tools today.

As requested, Clippy version: `clippy 0.0.212 (ea85b4c 2020-02-01)`

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #5154

changelog: Fix false positive in `single_component_path_imports`

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
@bors r+
thanks!

File: clippy_lints/src/single_component_path_imports.rs [hunk 1]
```diff
@@ -39,6 +39,7 @@ declare_lint_pass!(SingleComponentPathImports => [SINGLE_COMPONENT_PATH_IMPORTS]
 impl EarlyLintPass for SingleComponentPathImports {
     fn check_item(&mut self, cx: &EarlyContext<'_>, item: &Item) {
         if_chain! {
+            if !in_macro(item.span);
```
Comment (krishna-veerareddy) [inline clippy_lints/src/single_component_path_imports.rs]: Shouldn't we also exclude external macros?

Comment (JohnTitor) [reply clippy_lints/src/single_component_path_imports.rs]: `in_macro` should ignore any macros including external macros.

Comment (krishna-veerareddy) [reply clippy_lints/src/single_component_path_imports.rs]: Ah gotcha! I thought it was meant only for internal macros. My bad.

**Pull request merged**